### ISSUE TITLE
subscript out of bounds

### ISSUE THREAD

**Issue by abiyug** [rstudio/shiny#1199]
I am trying to run a reactive shiny app, and keep getting 'subscription out of bound' error.
```
Warning: Error in .subset2: subscript out of bounds
Stack trace (innermost first):
    86: reactiveValuesToList
    85: as.data.frame
```
Tried to get answer from the community on google and stackoverflow, but no answers. And I am reading that this maybe a bug on renderTable or shiny. Here is the full code to recreate.

```
library(shiny)
library(stats)

ui     <- fluidPage(
        h1("Temp Predictor for US Mainland"),
        sidebarLayout(
                sidebarPanel(
                        numericInput(
                                "new_lat", "Enter latitude between 20 and 60:", 21
                        )
                ),
                mainPanel(
                        plotOutput("table")
                ) 
        )
)

server <- function(input, output, session) {
        load("regression.lm.RData")
        lat <- reactive({
                 as.data.frame(reactiveValuesToList(input$new_lat))
                       })
        output$table <- renderTable(
                        predict(regression.lm, lat())
        )
}

shinyApp(ui, server)
```

**Comment by bborgesr** [comment]
It looks like the problem is the way you're using `reactiveValuesToList()`. You cannot pass `input$new_lat` on to `reactiveValuesToList()` because `input$new_lat` is not a reactivevalues object. You can check this yourself by running `is.reactivevalues(input$new_lat)` and printing the result somewhere inside your server function.

But, in any case, I'm a little confused as to what exactly are you trying to do there? `input$new_lat` is a number -- are you trying to convert it to a 1x1 dataframe? Or do you want to replicate that number a certain number of times (A) to get Ax1 or 1xA or even AxA? In any of these cases, you absolutely don't need `reactiveValuesToList`. In fact, if all you want is to generate  a 1x1 dataframe then `as.data.frame(input$new_lat)` is all you need.

A side note: you cannot use `renderTable` for a `plotOutput` slot. It looks to me like what you really want is a table, so you should change `plotOutput` to `tableOutput`. Otherwise, no table will appear.

**Comment by abiyug** [comment]
Barbara,
I saturated with what you have recommended `as.data.frame(input$new_lat)` also changed tableOutput on the UI.  I get error `object 'lat' not found`, shiny error's messages are not really that helpful.  

Let me clarify what the objective is:
I want to use a linear regression model to predict the temperature given latitude variable. [Here is my blog](http://abiyug.github.io/2016-05-24-Predictive-Modeling-v1) that describes this in detail.  

In short - 
1. I have a linear model        `regression.lm <- lm(apr ~ lat)`
2. I use the `predict` function to get new temp value - `pred <- predict(regression.lm, new_lat)`
3. From R console I create a data frame   `new_lat <- data.frame(lat = c(25, 55))`  
and this works.

I want the user to be able to enter value for `lat` and that value should be used in the predict function. Hope this clarifies, and thanks again.

Abiyu

**Comment by bborgesr** [comment]
The following code works fine:

```r
library(shiny)
library(stats)

ui     <- fluidPage(
  h1("Temp Predictor for US Mainland"),
  sidebarLayout(
    sidebarPanel(
      numericInput(
        "new_lat", "Enter latitude between 20 and 60:", 21
      )
    ),
    mainPanel(
      tableOutput("table")
    ) 
  )
)

server <- function(input, output, session) {
  lat <- reactive({
    as.data.frame(input$new_lat)
  })
  output$table <- renderTable({
    x <- rnorm(15)
    y <- x + rnorm(15)
    regression.lm <- lm(y ~ x)
    predict(regression.lm, lat())
  })
}

shinyApp(ui, server)
```

I don't have access to `regression.lm.RData`, so I made up a simple regression based on arbitrary values. But it should apply to whatever you have. My question has about the dimension of `lat()` itself. Is it supposed to be a 1x1 matrix, like in my example above? Or is it supposed to have other dimensions?

**Comment by abiyug** [comment]
1) lat value  is a 1x1 matrix.

2) you can grab the `regression.lm.RData from [my git hub ripository](https://github.com/abiyug/regression).

3) When I run your code, I get this error 
`no applicable method for 'xtable' applied to an object of class "c('double', 'numeric')" `

4) Another question that I am not clear on is that if you had saved a model, where do you call it in the shiny app? At the start of the server code inside of `{`? or inside the render table?

**Comment by abiyug** [comment]
Any update?

**Comment by abiyug** [comment]
Can some on kindly provide guidance on this issue?

**Comment by wch** [comment]
@abiyug The code provided by @bborgesr will run on the latest development version of Shiny, but in the current release version of Shiny, I see the issue you mentioned. Here's a slightly modified version:

``R
library(shiny)
library(stats)

ui     <- fluidPage(
  h1("Temp Predictor for US Mainland"),
  sidebarLayout(
    sidebarPanel(
      numericInput(
        "new_lat", "Enter latitude between 20 and 60:", 21
      )
    ),
    mainPanel(
      tableOutput("table")
    ) 
  )
)

server <- function(input, output, session) {
  lat <- reactive({
    data.frame(x = input$new_lat)
  })
  output$table <- renderTable({
    x <- rnorm(15)
    y <- x + rnorm(15)
    regression.lm <- lm(y ~ x)
    predict(regression.lm, lat())
  })
}

shinyApp(ui, server)
```

For (4), you can load it at the top of the server function.

This is a general help question, not a Shiny bug, and the best place to ask those questions is the shiny-discuss list.

**Comment by abiyug** [comment]
Thanks Winston for your input.  With your touch on the data.frame and using the `textOutput() - renderText({})` instead of `tableoutput/rendertable` made it work!
